build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -159,6 +206,13 @@ export function toReportKind(
 			break;
 		}
 
+		case "legacy-compat": {
+			for (const [pkg, details] of Object.entries(report)) {
+				toReturn[pkg] = details.ranges.legacyCompat;
+			}
```

Comment:
This is the place where I would check the release groups and adjust the version ranges if needed. If the release group of the package isn't client, then you can use the caret range (and you don't even have to calculate it - its already there on the details object.

Ideally though the client wouldn't be hardcoded. That said, I don't have a better option at the moment.